api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -450,6 +450,30 @@ type FelixConfigurationSpec struct {
 	// set to false. This reduces the number of metrics reported, reducing Prometheus load. [Default: true]
 	PrometheusWireGuardMetricsEnabled *bool `json:"prometheusWireGuardMetricsEnabled,omitempty"`
 
+	// MetricsTLSEnabled specifies whether TLS encryption is enabled for the /metrics endpoint.
+	// If set to true, the metrics server will only be accessible over HTTPS. Default is false.
+	MetricsTLSEnabled *bool `json:"metricsTLSEnabled,omitempty"`
+
+	// MetricsTLSCACertFile defines the absolute path to the TLS CA certificate file used for securing the /metrics endpoint.
+	// This certificate must be valid and accessible by the calico-node process.
+	MetricsTLSCACertFile *string `json:"metricsTLSCACertFile,omitempty"`
+
+	// MetricsTLSCertFile defines the absolute path to the TLS certificate file used for securing the /metrics endpoint.
+	// This certificate must be valid and accessible by the calico-node process.
+	MetricsTLSCertFile *string `json:"metricsTLSCertFile,omitempty"`
+
+	// MetricsTLSPrivateKeyFile defines the absolute path to the private key file corresponding to the TLS certificate
+	// used for securing the /metrics endpoint. The private key must be valid and accessible by the calico-node process.
+	MetricsTLSPrivateKeyFile *string `json:"metricsTLSPrivateKeyFile,omitempty"`
```

Comment:
```suggestion
	// PrometheusMetricsCertFile is the path to the TLS certificate file for the Prometheus metrics server. [Default: empty]
	PrometheusMetricsCertFile string `json:"prometheusMetricsCertFile,omitempty"`
	// PrometheusMetricsKeyFile is the path to the TLS private key file for the Prometheus metrics server. [Default: empty]
	PrometheusMetricsKeyFile string `json:"prometheusMetricsKeyFile,omitempty"`
	// PrometheusMetricsCAFile is the path to the TLS CA file for the Prometheus metrics server. [Default: empty]
	PrometheusMetricsCAFile string `json:"prometheusMetricsCAFile,omitempty"`
```

^ This aligns these fields with equivalent naming already used in our Enterprise version of Calico.